client/components/data/query-products-list/index.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { requestProductsList } from 'calypso/state/products-list/actions';
+import { isProductsListFetching, getProductsListType } from 'calypso/state/products-list/selectors';
+import type { CalypsoDispatch } from 'calypso/state/types';
+import type { AppState } from 'calypso/types';
+
+type QueryProductsListProps = {
+	type?: string;
+	currency?: string;
+	persist?: boolean;
+	productSlugList?: string[];
+};
+
+type RequestProps = Omit< QueryProductsListProps, 'productSlugList' > & {
+	product_slugs?: string[];
+};
```

Comment:
This complication is needed only because there are two different names for the same thing: `productSlugList` and `product_slugs`. Can we unify the code to use `productSlugList` everywhere?

Or, alternatively, define two different types that are similar, but not the same. `RequestProps` will have mandatory `type` field, and a `product_slugs` field. It will be exactly the object that is then passed as `query` to the REST request.

Reply:
Makes sense, updated.